### PR TITLE
[ SPARK-20739][CORE][TEST]Supplement the new unit tests to SparkContextSchedulerCreationSuite

### PR DESCRIPTION
**Pull request by heary-cao**
## What changes were proposed in this pull request?

This PR adds the new unit tests to support test 'SPARK_REGEX(sparkUrl)' branch in createTaskScheduler methods.  

## How was this patch tested?

The new unit test.

### CODE REVIEW

**Comment by srowen** [comment]
OK. It's minor, and not 100% sure this is the best place to test, but, it follows the pattern in the rest of the file and. This didn't need a JIRA, and please see my comments on the JIRA.

File: core/src/test/scala/org/apache/spark/SparkContextSchedulerCreationSuite.scala [hunk 1]
```diff
@@ -129,4 +129,29 @@ class SparkContextSchedulerCreationSuite
       case _ => fail()
     }
   }
+
+  test("bad-spark-url") {
+    val e = intercept[SparkException] {
+      createTaskScheduler("spark:*")
```
Comment (sameeragarwal) [inline core/src/test/scala/org/apache/spark/SparkContextSchedulerCreationSuite.scala]: Aren't these tests identical to existing tests above (`bad-master` and `local-default-parallelism`)?

**Comment by sameeragarwal** [comment]
@heary-cao it seems like the additional tests are primarily targeted towards testing [spark regex parsing](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L2731-L2736) while creating a task scheduler . If that's actually the intent, maybe we should have a more surgical test?

File: core/src/test/scala/org/apache/spark/SparkContextSchedulerCreationSuite.scala (hunk 1 above)
Comment (heary-cao) [inline core/src/test/scala/org/apache/spark/SparkContextSchedulerCreationSuite.scala]: The intent of this test is the spark regex parsing analysis anomalies.

**Comment by heary-cao** [comment]
Sorry,
surgical test? 
I didn't understand .
Can you explain in more detail?

**Comment by srowen** [comment]
He's saying that you're really just testing arg parsing. And it's mostly tested elsewhere, although maybe not this exact case. I don't know if this adds much value.

**Comment by heary-cao** [comment]
ok,
It does not produce value, that is junk code.
close this PR.

**Pull request closed without merging**